packages/pangea-sdk/src/main/java/cloud/pangeacyber/pangea/audit/models/IEvent.java

```diff
@@ -0,0 +1,40 @@
+package cloud.pangeacyber.pangea.audit.models;
+
+import cloud.pangeacyber.pangea.exceptions.PangeaException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+
+public interface IEvent {
+	public void setTenantID(String tenantID);
+
+	public String getTenantID();
+
+	public static String canonicalize(IEvent event) throws JsonProcessingException {
+		//FIXME: Check what happend if there is nested maps, do they order keys also? We DO NOT want that
+		ObjectMapper mapper = JsonMapper
+			.builder()
+			.configure(MapperFeature.SORT_PROPERTIES_ALPHABETICALLY, true)
+			.build();
+		return mapper.writeValueAsString(event);
+	}
+
+	public static <EventType extends IEvent> IEvent fromRaw(Object raw, Class<EventType> eventClass)
+		throws PangeaException {
+		if (raw == null) {
+			System.out.println("Event raw is null");
```

Comment:
should we be using System.out ever? Shoudnt we be accepting a logger object and then logging to that object?

Reply:
No, for sure, you are right, it was for debug. I'll check a proper way to add a logger object. Any idea is welcome! @ggallien13 @MichaelCombs28 feel free to give me your thoughts about it